x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapperTests.java

```diff
@@ -491,6 +467,79 @@ public void testMissingTaskType() throws IOException {
         assertThat(ex.getCause().getMessage(), containsString("failed to parse field [model_settings]"));
     }
 
+    public void testExistsQuerySparseVector() throws IOException {
+        final String fieldName = "semantic";
+        final String inferenceId = "test_service";
+
+        MapperService mapperService = mapperServiceForFieldWithModelSettings(
+            fieldName,
+            inferenceId,
+            new SemanticTextField.ModelSettings(TaskType.SPARSE_EMBEDDING, null, null)
+        );
+
+        Mapper mapper = mapperService.mappingLookup().getMapper(fieldName);
+        assertNotNull(mapper);
+        Query existsQuery = ((SemanticTextFieldMapper) mapper).fieldType().existsQuery(createSearchExecutionContext(mapperService));
+        assertThat(existsQuery, instanceOf(ESToParentBlockJoinQuery.class));
```

Comment:
Can you add back the inner query check after this?

Reply:
I was thinking that's too much coupling and details - it's not about checking the bool query with two clauses that are resulting from this, but the actual need to do so given we have YAML coverage. We're functionally covered, and we probably shouldn't care about the actual details of the query rewriting.

Happy to add the checks if you think it's absolutely needed

Reply:
I agree that hard-coding the expected query class creates too much coupling. However, as written `testExistsQuerySparseVector` & `testExistsQueryDenseVector` are not meaningfully different. In particular, the lack of assertions on the child query means that we're not testing that the child query is reflective of the backing embedding field type (`sparse_vector` or `dense_vector`).

I thought about how to resolve this without creating unmaintainable couplings and came up with this approach:

```
MapperBuilderContext mapperBuilderContext = MapperBuilderContext.root(false, false);
SearchExecutionContext searchExecutionContext = createSearchExecutionContext(mapperService);
SparseVectorFieldMapper sparseVectorFieldMapper = new SparseVectorFieldMapper.Builder(SemanticTextField.getEmbeddingsFieldName(fieldName)).build(mapperBuilderContext);

...

assertThat(toParentBlockJoinQuery.getChildQuery(), equalTo(sparseVectorFieldMapper.fieldType().existsQuery(searchExecutionContext)));
```

And something similar for `testExistsQueryDenseVector`, except creating a `DenseVectorFieldMapper`. WDYT?

Reply:
I like that, I think it's a good middle ground!

Reply:
@Mikep86 I've included additional checks in cabf436bff2f784a48cacb9d080760f43ce94ff5.

I'm not convinced on this - we're essentially re-doing what the implementation does internally. But I couldn't find another way of including the nested path in the final bool query otherwise.